File: I1_MVPAndRetrofit/app/src/main/java/com/example/tsvetelinastoyanova/weatherreportrevisited/citieslist/CitiesListContract.java
package com.example.tsvetelinastoyanova.weatherreportrevisited.citieslist;

import com.example.tsvetelinastoyanova.weatherreportrevisited.BasePresenter;
import com.example.tsvetelinastoyanova.weatherreportrevisited.BaseView;
import com.example.tsvetelinastoyanova.weatherreportrevisited.City;
import com.example.tsvetelinastoyanova.weatherreportrevisited.data.CityEntity;
import com.example.tsvetelinastoyanova.weatherreportrevisited.model.WeatherObject;

import java.util.List;

public interface CitiesListContract {

    interface View extends BaseView<Presenter> {
        void showNewCityAdded(City newCity);

        void showErrorAddingAddedCity();

        void showErrorLoadingCities();

        void showCityDeleted();

        void showErrorDeleteCity();

        void showCityLoaded(City city);

        void showCityUpdated(City city);

        void setWeatherObjectWhenClicked(String cityName);

        List<City> getDisplayedCities();
    }

    interface Presenter extends BasePresenter {
        void loadCities();

        void addNewCity(String s);

    //    void updateCity(CityEntity s);

        void deleteCity(City city);

        void refreshCities(List<City> cities);

        WeatherObject getWeatherObjectOnClick(String cityName);

    //    void addNewWeatherObject(WeatherObject weatherObject);
    }
}

File: I1_MVPAndRetrofit_Kotlin/app/src/main/java/com/example/tsvetelinastoyanova/weatherapp/citiesList/CitiesListContract.kt
package com.example.tsvetelinastoyanova.weatherapp.citiesList

import android.support.annotation.VisibleForTesting
import com.example.tsvetelinastoyanova.weatherapp.BasePresenter
import com.example.tsvetelinastoyanova.weatherapp.BaseView
import com.example.tsvetelinastoyanova.weatherapp.City
import com.example.tsvetelinastoyanova.weatherapp.model.currentweather.CurrentWeatherObject

interface CitiesListContract {

    interface View : BaseView<Presenter> {

        fun showNewCityAdded(newCity: City)

        fun showErrorInternetConnection()

        fun showErrorAddingAddedCity()

        fun showErrorLoadingCities()

        fun showCityDeleted()

        fun showErrorDeleteCity()

        fun showErrorNotValidCityName()

        fun showCityLoaded(city: City)

        fun showCityUpdated(city: City)

        fun setWeatherObjectWhenClicked(cityName: String)

        fun getDisplayedCities(): MutableList<City>?

        @VisibleForTesting
        fun getCityFromInputField(): String
    }

    interface Presenter : BasePresenter {

        fun loadCities()

        fun addNewCity(s: String?)

        fun deleteCity(city: City)

        fun refreshCities(cities: MutableList<City>?)

        fun getWeatherObjectOnClick(cityName: String): CurrentWeatherObject?
    }
}